docs/release/generate_release_notes.py

```diff
@@ -78,25 +98,59 @@ def tqdm(i, **kwargs):
 else:
     raise RuntimeError(f'Desired tag ({args.from_commit}) not found')
 
+common_ancestor = local_repo.merge_base(args.to_commit, args.from_commit)[0]
+remote_commit = repository.get_commit(common_ancestor.hexsha)
+
 # For some reason, go get the github commit from the commit to get
 # the correct date
-github_commit = previous_tag.commit.commit
 previous_tag_date = datetime.strptime(
-    github_commit.last_modified, '%a, %d %b %Y %H:%M:%S %Z'
+    remote_commit.last_modified, '%a, %d %b %Y %H:%M:%S %Z'
 )
 
 
+def get_commits_to_ancestor(ancestor, rev="main"):
+    yield from local_repo.iter_commits(f'{ancestor.hexsha}..{rev}')
+
+
+new_commits_count = (
+    len(list(get_commits_to_ancestor(common_ancestor, args.to_commit))) + 1
+)
+release_branch_count = (
+    len(list(get_commits_to_ancestor(common_ancestor, args.from_commit))) + 1
+)
```

Comment:
For proper working, the local main need to be the same as the remote one. But maybe one knows how to specify remote labels?